src/sentry/tasks/delete_pending_groups.py

```diff
@@ -34,30 +35,43 @@ def delete_pending_groups() -> None:
     and schedules deletion tasks for them. Groups are batched by project to ensure
     efficient deletion processing.
 
-    Only processes groups with last_seen between 6 hours and 90 days ago to avoid
-    processing very recent groups (safety window) or groups past retention period.
+    Only processes groups where deletion was requested between 6 hours and 90 days ago
+    (via audit logs) to avoid processing very recent groups (safety window) or groups
+    past retention period.
     """
     statuses_to_delete = [GroupStatus.PENDING_DELETION, GroupStatus.DELETION_IN_PROGRESS]
 
     # Just using status to take advantage of the status DB index
-    groups = Group.objects.filter(status__in=statuses_to_delete).values_list(
-        "id", "project_id", "last_seen"
-    )[:BATCH_LIMIT]
-
-    if not groups:
-        logger.info("delete_pending_groups.no_groups_found")
-        return
+    groups = list(
+        Group.objects.filter(status__in=statuses_to_delete).values_list(
+            "id", "project_id", "last_seen"
+        )
+    )
 
-    # Process groups between 6 hours and 90 days old
     now = timezone.now()
     min_last_seen = now - timedelta(days=MAX_LAST_SEEN_DAYS)
-    max_last_seen = now - timedelta(hours=MIN_LAST_SEEN_HOURS)
+    cutoff_time = now - timedelta(hours=MIN_LAST_SEEN_HOURS)
+
+    # Query audit logs to find groups where deletion was requested in the time range
+    # (between 90 days ago and 6 hours ago)
+    non_recent_deletion_group_ids = set(
+        log_rpc_service.find_issue_deletions_before(
```

Comment:
Using an RPC call in the same change that the RPC method is added in can result in failures, as regions are deployed many minutes before control is.  You could put this call behind an option that is enabled once all deploys are done, or deploy the RPC method first, and then deploy the call site separately.